kong/router/atc.lua

```diff
@@ -724,36 +626,22 @@ function _M:select(_, _, _, scheme,
                       dst_ip, dst_port,
                       sni)
 
-  local c = context.new(self.schema)
-
-  for _, field in ipairs(self.fields) do
-    if field == "net.protocol" then
-      assert(c:add_value(field, scheme))
+  local match_params = assert(table_fetch(ATC_MATCH_PARAMS_POOL, 0, 6))
```

Comment:
Is table pool needed here? It doesn't looks like we need yielding. https://github.com/Kong/kong/blob/master/kong/tools/request_aware_table.lua may be enough.

Reply:
done, using module level table.